src/github.ts

```diff
@@ -328,4 +329,34 @@ export class GitHub {
     core.info(`Writing summary`);
     await sum.addSeparator().write();
   }
+
+  public static async annotateBuildWarnings(source: string, warnings?: Array<VertexWarning>): Promise<void> {
+    (warnings ?? []).forEach(warning => {
+      if (!warning.detail || !warning.short) {
+        return;
+      }
+      const title = warning.detail.map(encoded => atob(encoded)).join(' ');
+      let message = atob(warning.short).replace(/\s\(line \d+\)$/, '');
+      if (warning.url) {
+        // https://github.com/docker/buildx/blob/d8c9ebde1fdcf659f1fa3efa6ccc27a28b0f1564/commands/build.go#L854
+        message += `\nMore info: ${warning.url}`;
+      }
+      let startLine: number | undefined, endLine: number | undefined;
+      for (const range of warning.range ?? []) {
+        if (range.start.line && (!startLine || range.start.line < startLine)) {
+          startLine = range.start.line;
+        }
+        if (range.end.line && (!endLine || range.end.line > endLine)) {
+          endLine = range.end.line;
+        }
+      }
+      core.info(`${message} --- startLine: ${startLine}, endLine: ${endLine}`);
+      core.warning(message, {
+        title: title,
+        file: source,
+        startLine: startLine,
+        endLine: endLine
+      });
```

Comment:
Was looking at showing first and last lines matching warning ranges but GitHub annotations does not seem to display that properly:

![image](https://github.com/docker/actions-toolkit/assets/1951866/c5a43dd3-ad2d-427a-8fc7-c673582569b0)

`core.info` returns the right `startLine` and `endLine` through:

```
ConsistentInstructionCasing: Command 'COPy' should match the case of the command majority (lowercase)
More info: https://docs.docker.com/go/dockerfile/rule/consistent-instruction-casing/ --- startLine: 21, endLine: 23
```

I will just set `startLine` for now and open a thread on GitHub community.